src/vma/sock/sockinfo.cpp

```diff
@@ -107,6 +107,11 @@ sockinfo::sockinfo(int fd):
 	m_p_socket_stats->fd = m_fd;
 	m_p_socket_stats->inode = fd2inode(m_fd);
 	m_p_socket_stats->b_blocking = m_b_blocking;
+	m_p_socket_stats->ring_alloc_logic_rx = m_ring_alloc_log_rx.get_ring_alloc_logic();
+	m_p_socket_stats->ring_alloc_logic_tx = m_ring_alloc_log_tx.get_ring_alloc_logic();
+	m_p_socket_stats->ring_user_id_rx = m_ring_alloc_logic.calc_res_key_by_logic();
+	m_p_socket_stats->ring_user_id_tx =
+			ring_allocation_logic_tx(get_fd(), m_ring_alloc_log_tx, this).calc_res_key_by_logic();
```

Comment:
User id should be set from setsockopt. we should we update it from the constructor ?

Reply:
There's always a key/user_id to the ring - whether user set it or we got it from calculating the key value according to the core/thread/fd... This is initializing it to the initial value that is generated according to the default ring allocation logic.